tests/test_babamul.rs

```diff
@@ -128,32 +134,36 @@ async fn consume_kafka_messages(
         .expect("Failed to subscribe to topic");
 
     let mut messages = Vec::new();
-    let timeout = Duration::from_secs(5);
+    let timeout = Duration::from_secs(8);
     let start = std::time::Instant::now();
 
     let mut nb_errors = 0;
     let max_nb_errors = 5;
     while messages.len() < expected_count && start.elapsed() < timeout {
-        match consumer.recv().await {
-            Ok(message) => {
+        match tokio::time::timeout(timeout - start.elapsed(), consumer.recv()).await {
```

Comment:
that's an odd syntax? Kafka should have a built it equivalent to `recv()` that takes a timeout (at least that's what I've used most of the time in python, and I believe in rust too). Worth comparing this against the kafka consumer code?